yew/src/agent/local/context.rs

```diff
@@ -1,21 +1,28 @@
 use super::*;
 use crate::callback::Callback;
 use crate::scheduler::Shared;
+use crate::agent::SyncAgent;
 use anymap::{self, AnyMap};
 use slab::Slab;
 use std::cell::RefCell;
 use std::rc::Rc;
+use std::marker::PhantomData;
 
 thread_local! {
     static LOCAL_AGENTS_POOL: RefCell<AnyMap> = RefCell::new(AnyMap::new());
 }
 
 /// Create a single instance in the current thread.
 #[allow(missing_debug_implementations)]
-pub struct Context;
+pub struct Context<AGN> {
+    _agent: PhantomData<AGN>
+}
+
+impl<AGN: SyncAgent> Discoverer for Context<AGN> {
```

Comment:
Unfortunately the trait's associated type is not enough to constraint the generic type of `AGN`, so we need to modify the `Context` to carry that information. Otherwise this would be a change with no effect for library users.